File: src/main/java/nl/hand/made/vending/machine/repository/model/ProductEntity.java
package nl.hand.made.vending.machine.repository.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Getter
@Setter
@Entity
@Table(name = "PRODUCT")
public class ProductEntity {

    @NotNull
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    @Column(name = "AMOUNT_AVAILABLE", nullable = false)
    private Integer amountAvailable;

    @NotNull
    @Column(name = "COST", nullable = false)
    private Integer cost;

    @NotNull
    @Column(name = "PRODUCT_NAME", nullable = false)
    private String productName;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "SELLER_NAME", referencedColumnName = "USERNAME")
    private UserEntity seller;
}

File: src/main/kotlin/nl/hand/made/vending/machine/repository/model/ProductEntity.kt
package nl.hand.made.vending.machine.repository.model

import javax.persistence.*

@Entity
@Table(name = "PRODUCT")
class ProductEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Int? = null,
    var amountAvailable: Int,
    var cost: Int,
    var productName: String,

    @ManyToOne @JoinColumn(name = "SELLER_NAME", referencedColumnName = "USERNAME")
    var seller: UserEntity,
)
